### PR TITLE
Orca NCF example

### PR DESCRIPTION
**Pull request by jenniew**
Add Orca ncf tf example using xshards and tf estimator.

### CODE REVIEW

**Comment by jenniew** [comment]
Add both python script version and notebook version.

File: pyzoo/zoo/examples/orca/learn/tf/ncf.py [hunk 1]
```diff
@@ -0,0 +1,281 @@
...
+
+        with tf.name_scope("optimzation"):
+            self.optim = tf.train.AdamOptimizer(1e-3, name='Adam')
+            self.optimizer = self.optim.minimize(self.loss)
+
+
+def train(train_data, test_data, user_size, item_size):
+        model = NCF(opt.embedding_size, user_size, item_size)
+
+        estimator = zoo.orca.learn.tf.estimator.Estimator.from_graph(
```
Comment (jason-dai) [inline pyzoo/zoo/examples/orca/learn/tf/ncf.py:203]: change to Estimator.from_graph

File: pyzoo/zoo/examples/orca/learn/tf/ncf.py [hunk 2]
```diff
@@ -0,0 +1,281 @@
...
+    tf.reset_default_graph()
+
+    with tf.Session() as sess:
+        model = NCF(opt.embedding_size, user_size, item_size)
+
+        saver = tf.train.Saver(tf.global_variables())
+        checkpoint_path = os.path.join(opt.model_dir, "NCF.ckpt")
+        saver.restore(sess, checkpoint_path)
+
+        estimator = zoo.orca.learn.tf.estimator.Estimator.from_graph(
```
Comment (jason-dai) [inline pyzoo/zoo/examples/orca/learn/tf/ncf.py:240]: change to Estimator.from_graph

Comment (jenniew) [reply pyzoo/zoo/examples/orca/learn/tf/ncf.py]: udpated

File: pyzoo/zoo/examples/orca/learn/tf/ncf.py (hunk 1 above)
Comment (jenniew) [reply pyzoo/zoo/examples/orca/learn/tf/ncf.py]: updated

**Comment by jenniew** [comment]
Jenkins pass: http://10.239.47.210:18888/job/ZOO-PR-Validation/4340/

**Pull request merged**